cli/tests/pcluster/models/test_s3_bucket.py

```diff
@@ -301,3 +304,241 @@ def test_upload_file(mocker, content, file_name, file_type, s3_file_format, expe
+    )
+
+    partition = "fake_partition"
+    region = "fake-region"
+    account_id = "fake-id"
+    bucket_name = bucket.name
+
+    bucket_arn = format_arn(partition, "s3", "", "", bucket_name)
+    bucket_arn_with_wildcard = f"{bucket_arn}/*"
+    logs_service_principal = get_service_principal(
+        service_name="logs", partition=partition, region=region, regional=True
+    )
+
+    log_group_arns = [
+        format_arn(partition, "logs", region, account_id, "log-group:/aws/parallelcluster/*"),
+        format_arn(partition, "logs", region, account_id, "log-group:/aws/imagebuilder/*"),
+    ]
+
+    policy_statements = bucket._get_bucket_policy_for_cloudwatch_logs()
+
+    assert_that(policy_statements).is_length(3)
+
+    statements_by_sid = {statement["Sid"]: statement for statement in policy_statements}
+
+    # Verify AllowReadBucketAclForExportLogs policy
+    read_acl_statement = statements_by_sid.get("AllowReadBucketAclForExportLogs")
+    assert_that(read_acl_statement).is_not_none()
+    assert_that(read_acl_statement["Action"]).is_equal_to("s3:GetBucketAcl")
+    assert_that(read_acl_statement["Effect"]).is_equal_to("Allow")
+    assert_that(read_acl_statement["Resource"]).is_equal_to(bucket_arn)
+    assert_that(read_acl_statement["Principal"]).is_equal_to({"Service": logs_service_principal})
+    assert_that(read_acl_statement["Condition"]).is_equal_to(
+        {
+            "StringEquals": {"aws:SourceAccount": account_id},
+            "ArnLike": {"aws:SourceArn": log_group_arns},
+        }
+    )
+
+    # Verify AllowPutObjectForExportLogs policy
+    put_object_statement = statements_by_sid.get("AllowPutObjectForExportLogs")
+    assert_that(put_object_statement).is_not_none()
+    assert_that(put_object_statement["Action"]).is_equal_to("s3:PutObject")
+    assert_that(put_object_statement["Effect"]).is_equal_to("Allow")
+    assert_that(put_object_statement["Resource"]).is_equal_to(bucket_arn_with_wildcard)
+    assert_that(put_object_statement["Principal"]).is_equal_to({"Service": logs_service_principal})
+    assert_that(put_object_statement["Condition"]).is_equal_to(
+        {
+            "StringEquals": {
+                "s3:x-amz-acl": "bucket-owner-full-control",
+                "aws:SourceAccount": account_id,
+            },
+            "ArnLike": {"aws:SourceArn": log_group_arns},
+        }
+    )
+
+    # Verify DenyPutObjectOnReservedPath policy
+    deny_put_object_statement = statements_by_sid.get("DenyPutObjectOnReservedPath")
+    assert_that(deny_put_object_statement).is_not_none()
+    assert_that(deny_put_object_statement["Action"]).is_equal_to("s3:PutObject")
+    assert_that(deny_put_object_statement["Effect"]).is_equal_to("Deny")
+    assert_that(deny_put_object_statement["Resource"]).is_equal_to(f"{bucket_arn}/parallelcluster/*")
+    assert_that(deny_put_object_statement["Principal"]).is_equal_to({"Service": logs_service_principal})
+
+
+def test_generate_bucket_policy(mocker):
```

Comment:
Same as above: reduce the assertion to an assert equal to a static json instead of asserting field by field.

Reply:
Done!